solr/core/src/test/org/apache/solr/cloud/CollectionsAPISolrJTest.java

```diff
@@ -584,6 +591,7 @@ public void testCollectionProp() throws InterruptedException, IOException, SolrS
     final String propName = "testProperty";
 
     CollectionAdminRequest.createCollection(collectionName, "conf", 2, 2)
+        .setPerReplicaState(SolrCloudTestCase.USE_PER_REPLICA_STATE)
```

Comment:
Do we run these tests twice? Once with `USE_PER_REPLICA_STATE` `false` (as it the default) and one with it set to `true`?

I'm not seeing it set to `true` anywhere. Are you basically manually changing your environment to run these tests?

Reply:
Yes, We run these tests with that property `true` and `false`

We couldn't find a better way to do this yes

Reply:
Building it this way is fine. But for clarity what I meant was that the default value is false. And these tests will *not* be run by any CI system. Neither ours, nor Solrs. With the value set to `true`.

I think we either want to have these tests literally run these tests twice somehow. Or very clearly document which configuration we need to test appropriately.

I think of testing as a single suite testing all permutations. Not a runtime-sensitive thing. If I were settings this up with continuous tests, I would want to run these parameterized tests for each value of `USE_PER_REPLICA_STATE`.